src/ArduinoSpotify.cpp

```diff
@@ -757,10 +884,23 @@ void ArduinoSpotify::parseError()
     }
 }
 
-void ArduinoSpotify::closeClient()
+void ArduinoSpotify::stopClient(bool force)
```

Comment:
This looks interesting, I think one potential issue we might have with it is there two different servers, one for spotify api and one for images. Probably could just be handled at the sketch level, before you request the image server you could do a `stopClient(true);`

Reply:
oh, I totally missed that one and now I am wondering why/how this still works for me 🤔

Reply:
I don't think its a big deal to do it at the sketch level:

I'm just testing it out now.

Also if you want to merge in the latest master I fixed the failing test (or at least I should have, no big deal either way though)

Reply:
The ESP8266 doesn't seem to have enough heap memory to do the keep-alive stuff, will add some more debug messages to see what exactly is happening, but it doesn't have enough memory to parse the JSON it seems

Setting the keepAlive flag to false is causing a crash on now, looking into it

Reply:
Not sure the benefit outweights the problems this might cause. If the user has to take care of closing the connection, this doesn't seem too easy to use.
Also the benefit isn't that big anymore, if the connection has to be reopened in many cases anyway. (I was mainly aiming for the token refresh to improve, which won't work, since the account host is also a different one).

Also I figured out why it currently works: api.spotify.com and accounts.spotify.com resolve to the same IP address (at least for me), so the connection can be reused (by coincidence), but probably shouldn't.

Maybe we should at least disable the useHttpKeepAlive by default?

Reply:
Yeah possibly.

If i set the keep alive to false it is still crashing for me on a sample sketch I have (Display album art on a TFT screen)

`
Decoding stack results
0x4010026a: millis() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/core_esp8266_wiring.cpp line 188
0x402109c6: optimistic_yield(uint32_t) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/core_esp8266_main.cpp line 146
0x4020308f: WiFiClient::available() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/WiFiClient.cpp line 263
0x40204ae6: BearSSL::WiFiClientSecure::_run_until(unsigned int, bool) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp line 527
0x402317d6: br_ssl_engine_hs_reset at src/ssl/ssl_engine.c line 1320
0x40204d48: BearSSL::WiFiClientSecure::_wait_for_handshake() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp line 564
0x40204f2b: BearSSL::WiFiClientSecure::_connectSSL(char const*) at /Users/brian/Library/Arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/xtensa-lx106-elf/include/c++/4.8.2/bits/shared_ptr.h line 291
0x4021095c: __esp_yield() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/core_esp8266_main.cpp line 119
0x40210faa: __delay(unsigned long) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/core_esp8266_wiring.cpp line 54
0x40203ae4: WiFiClient::connect(IPAddress, unsigned short) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/include/ClientContext.h line 144
0x40205049: BearSSL::WiFiClientSecure::connect(char const*, unsigned short) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp line 232
0x4020965d: ArduinoSpotify::makeGetRequest(char const*, char const*, char const*, char const*) at /Users/brian/Documents/Arduino/libraries/arduino-spotify-api/src/ArduinoSpotify.cpp line 133
0x4020afb5: ArduinoSpotify::getCurrentlyPlaying(char const*) at /Users/brian/Documents/Arduino/libraries/arduino-spotify-api/src/ArduinoSpotify.cpp line 560
0x40212549: uart_write(uart_t*, char const*, size_t) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/uart.cpp line 509
0x4020308f: WiFiClient::available() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi/src/WiFiClient.cpp line 263
0x4020f97c: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 164
0x4020f988: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 165
0x4020f97c: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 164
0x4020fc49: Print::write(char const*) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/Print.h line 62
0x4020fdf9: Print::printNumber(unsigned long, unsigned char) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/Print.cpp line 268
0x40212549: uart_write(uart_t*, char const*, size_t) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/uart.cpp line 509
0x40212549: uart_write(uart_t*, char const*, size_t) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/uart.cpp line 509
0x4010026a: millis() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/core_esp8266_wiring.cpp line 188
0x4020f97c: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 164
0x4020f988: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 165
0x4020f97c: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/HardwareSerial.h line 164
0x4020fc49: Print::write(char const*) at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/Print.h line 62
0x4020fd10: Print::println() at /Users/brian/Library/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/Print.cpp line 186
0x402023b0: loop() at /Users/brian/Documents/Arduino/test/test.ino line 325`

From a debugging point of view it is a little awkward cause i'm on a different machine as well as using the PR code so i'm not sure whats at fault yet.

Reply:
The "Connection: Close" header seems to be a problem - no idea why, but shouldn't hurt to remove it.

Reply:
While digging deeper I found a few more (potential) issues and fixed them, but I guess I'll drop the keep-alive thing, as it adds to much complexity.
Maybe I'll give it another try later on using the existing HTTPClient implementations, but that doesn't have to be mixed up with the getDevices call